components/sup/src/manager/service/config.rs

```diff
@@ -170,8 +215,54 @@ impl Cfg {
         Ok(())
     }
 
-    fn load_environment(&mut self, package: &Pkg) -> Result<()> {
-        let var_name = format!("{}_{}", ENV_VAR_PREFIX, package.name)
+    fn open_user_toml<P: PackageForConfig>(&mut self, package: &P) -> Option<(File, PathBuf)> {
+        let recommended_path = package.user_config_path().join("user.toml");
+        let ucps = if let Some(ref ucp) = self.user_config_path {
+            // Marking the path as not deprecated, even if the user
+            // config path points to the old path. If this is the
+            // case, we already warned about it.
+            vec![UserConfigPath::Recommended(ucp.get_path().clone())]
+        } else {
+            vec![
+                UserConfigPath::Recommended(recommended_path.clone()),
+                UserConfigPath::Deprecated(package.svc_path().join("user.toml")),
+            ]
+        };
+        for ucp in ucps {
+            match File::open(ucp.get_path()) {
+                Ok(file) => {
+                    if let UserConfigPath::Deprecated(_) = ucp {
+                        outputln!(
+                            "The {} at {} is deprecated, consider putting it in {}",
+                            "user configuration location",
+                            ucp.get_path().display(),
+                            recommended_path.display(),
+                        );
```

Comment:
This format string has three placeholders, but only two values are supplied.

Reply:
It has three values, first is the `"user configuration location"` string. I did it that way, otherwise the line was over 100 chars long and rustfmt was bailing out… I'll split it out to a separate function so I'll have less indentation and more space for the string.

Reply:
Ah, you're right... I totally missed that.

You can also split long strings up, like we do here: https://github.com/habitat-sh/habitat/blob/cc079b61663b826a4da9e88939eaeaffdf3acb3e/components/sup/src/main.rs#L160-L162

Reply:
Ah, nice. That's exactly what I need. Thanks!